mrp_multi_level/models/product_mrp_area.py

```diff
@@ -205,20 +210,19 @@ def _get_rule(self):
         return rule
 
     def _compute_supply_method(self):
+        boms_by_product = self.env["mrp.bom"]._bom_find(self.mapped("product_id"))
         for rec in self:
             rule = rec._get_rule()
             if not rule:
                 rec.supply_method = "none"
-                continue
-            # Determine the supply method based on the final rule.
-            boms = rec.product_id.product_tmpl_id.bom_ids.filtered(
-                lambda x: x.type in ["normal", "phantom"]
-            )
-            rec.supply_method = (
-                "phantom"
-                if rule.action == "manufacture" and boms and boms[0].type == "phantom"
-                else rule.action
-            )
+                rec.supply_bom_id = False
+            elif rule.action == "manufacture":
```

Comment:
You are changing aditional the logic, BoM is applicable also in phanton and subcontracting for example, please keep the logic as it was without this new elif and else sentences and keep only the change regarding supply BoM and using the variant

Reply:
@LoisRForgeFlow  I pushed new code for this.  There is still a very small difference in the logic, but I can't find any reason it will be a problem.  It would be ugly to use the _bom_find() method, and match the original logic exactly.

Matching the original logic exactly, and using the _bom_find() method, would require one of the following:

- Call _bom_find() again (and potentially again), if BoM type not in ("normal", "phantom")
- Modify _bom_find_domain() and pass multiple BoM types in context
- Do a PR with core Odoo to make _bom_find_domain() accept a list of types

Reply:
Yes, I think now it is perfect, I don't see any big difference apart from the way the BoM is searched (which is now better)